File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        int start = 0, end = s.size()-1;
        while(start < end)
        {
            if(!isalnum(s[start]))
                start++;
            else if(!isalnum(s[end]))
                end--;
            else 
            {
                if(isalpha(s[start]) && isalpha(s[end]))
                {
                    char ch1 = (char)tolower(s[start]);
                    char ch2 = (char)tolower(s[end]);
                    if(ch1 == ch2)
                    {
                        start++;
                        end--;
                    }
                    else
                        return false;
                }
                else if(s[start] == s[end])
                    {
                        start++;
                        end--;
                    }
                else
                    return false;
            }
        }
        return true;
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        start, end = 0, len(s)-1
        while start < end:
            while start < end and s[start].isalnum() is False:
                start+=1
            while start < end and s[end].isalnum() is False:
                end-=1
            if start < end and s[start].lower()!= s[end].lower():
                return False
            start+=1
            end-=1
        return True